### PR TITLE
fix: toast renders on each save

### PR DESCRIPTION
**Pull request by AntBush**
Implements # 1529

### CODE REVIEW

File: app/components/Analyst/Project/Announcements/AnnouncementsForm.tsx [hunk 1]
```diff
@@ -194,6 +195,9 @@ const AnnouncementsForm = ({ query }) => {
   const removeSelfReference = (ccbcList: Array<any>) => {
     return ccbcList.filter((ccbcId) => ccbcId.ccbcNumber !== ccbcNumber);
   };
+  const handleUpdateToastKey = () => {
+    setToastKey((key) => key + 1);
+  };
 
   const handleSubmit = () => {
     hiddenSubmitRef.current.click();
```
Comment (marcellmueller) [inline app/components/Analyst/Project/Announcements/AnnouncementsForm.tsx:203]: Nice work on this toast, it looks great.

I think this solution is slightly overcomplicated. I think we can simply set `updatedCcbcItems` to null at the top of the `handleSubmit`?
```suggestion
  const handleSubmit = () => {
    setUpdatedCcbcItems(null);
    hiddenSubmitRef.current.click();
```

Comment (AntBush) [reply app/components/Analyst/Project/Announcements/AnnouncementsForm.tsx]: Worked perfectly! Much better than my convoluted method

**Comment by AntBush** [comment]
> I am getting an error on announcements that don't have any additional projects (ccbc numbers) added to them:
> 
> Screenshot 2023-05-15 at 8 17 17 AM

Fixed the error

**Review by marcellmueller — APPROVED** [review]
🚢

**Pull request merged**